packages/core/src/external/aws/lambda-logic/bundle-to-html-adhd.ts

```diff
@@ -0,0 +1,2339 @@
+                <a href="#immunizations">Immunizations</a>
+              </li>
+              <li>
+                <a href="#family-member-history">Family Member History</a>
+              </li>
+              <li>
+                <a href="#related-persons">Related Persons</a>
+              </li>
+              <li>
+                <a href="#tasks">Tasks</a>
+              </li>
+              <li>
+                <a href="#coverage">Coverage</a>
+              </li>
+              <li>
+                <a href="#encounters">Encounters</a>
+              </li>
+            </div>
+            </ul>
+        </div>
+      </div>
+    </div>
+  `;
+}
+
+function createHeaderTableRow(label: string, value: string) {
+  return `
+    <tr>
+      <td>
+        <span class="header-label">${label}</span>
+        <span>
+          ${value}
+        </span>
+      </td>
+    </tr>
+  `;
+}
+
+type EncounterTypes =
+  | "labs"
+  | "progressNotes"
+  | "afterInstructions"
+  | "reasonForVisit"
+  | "documentation";
+
+type EncounterSection = {
+  [key: string]: {
+    [k in EncounterTypes]?: DiagnosticReport[];
+  };
+};
+
+function createDiagnosticReportsSection(
+  diagnosticReports: DiagnosticReport[],
+  practitioners: Practitioner[],
+  aweAndADHDVisits: Condition[],
+  organizations: Organization[]
+) {
+  const mappedPractitioners = mapResourceToId<Practitioner>(practitioners);
+  const mappedOrganizations = mapResourceToId<Organization>(organizations);
+
+  if (!diagnosticReports) {
+    return "";
+  }
+
+  const encounterSections: EncounterSection = buildEncounterSections({}, diagnosticReports);
+
+  const encountersWithoutAWEAndADHD = Object.entries(encounterSections)
+    .filter(([key]) => {
+      const encounterDateFormatted = dayjs(key).format(ISO_DATE);
+      const leftOverVisit = aweAndADHDVisits.find(visit => {
+        const visitDate = visit.onsetDateTime ?? "";
+        const visitDateFormatted = dayjs(visitDate).format(ISO_DATE);
+
+        return visitDateFormatted === encounterDateFormatted;
+      });
+
+      return !leftOverVisit;
+    })
+    .reduce((acc, [key, value]) => {
+      acc[key] = value;
+      return acc;
+    }, {} as EncounterSection);
+
+  const aYearAgo = dayjs().subtract(1, "year").format(ISO_DATE);
+
+  const nonAWEreports = buildReports(
+    encountersWithoutAWEAndADHD,
+    mappedPractitioners,
+    mappedOrganizations,
+    aYearAgo
+  );
+
+  const hasNonAWEreports = nonAWEreports.length > 0;
+
+  return `
+    <div id="reports" class="section">
+      <div class="section-title">
+        <h3 id="reports" title="reports">&#x276F; Reports</h3>
+        <a href="#mr-header">&#x25B2; Back to Top</a>
+      </div>
+      <div class="section-content">
+        ${
+          hasNonAWEreports
+            ? nonAWEreports
+            : `<table><tbody><tr><td>No reports found</td></tr></tbody></table>`
+        }
+      </div>
+    </div>
+  `;
+}
+
+function createFilteredReportSection(
```

Comment:
This is the refactored report sections that can now handle any filter from a condition ie. adhd or awe